test/integration/testing.py

```diff
@@ -152,6 +156,127 @@ def _catalog_columns(self, use_catalog=False, use_surrogate=True):
 			cols_expect.sort()
 			self.assertEqual(cols_have, cols_expect)
 
+
+	# pyodbc doesn't support INTERVAL types; when installing an "output converter", it asks the ODBC driver for the
+	# binary format and currently, this is the same as a wchar_t string for INTERVALs.
+	# Also, just return None for data type 0 -- NULL
+	def _install_output_converters(self, cnxn):
+		wchar_sz = ctypes.sizeof(ctypes.c_wchar)
+		if wchar_sz == ctypes.sizeof(ctypes.c_ushort):
+			unit = "H"
+		elif wchar_sz == ctypes.sizeof(ctypes.c_uint32):
+			unit = "I"
+		else:
+			raise Exception("unsupported wchar_t size")
+
+		# wchar_t to python string
+		def _convert_interval(value):
+			cnt = len(value)
+			assert(cnt % wchar_sz == 0)
+			cnt //= wchar_sz
+			ret = ""
+			fmt = "=" + str(cnt) + unit
+			for c in struct.unpack(fmt, value):
+				ret += chr(c)
+			return ret
+
+		for x in range(101, 114): # INTERVAL types IDs
+			cnxn.add_output_converter(x, _convert_interval)
+
+		def _convert_null(value):
+			return None
+		cnxn.add_output_converter(0, _convert_null) # NULL type ID
+
+	# produce an instance of the 'data_type' out of the 'data_val' string
+	def _type_to_instance(self, data_type, data_val):
+		# Change the value read in the tests to type and format of the result expected to be
+		# returned by driver.
+		if data_type == "null":
+			instance = None
+		elif data_type.startswith("bool"):
+			instance = data_val.lower() == "true"
+		elif data_type in ["byte", "short", "integer"]:
+			instance = int(data_val)
+		elif data_type == "long":
+			instance = int(data_val.strip("lL"))
+		elif data_type == "double":
+			instance = float(data_val)
+		elif data_type == "float":
+			instance = float(data_val.strip("fF"))
+		elif data_type in ["datetime", "date", "time"]:
+			fmt = "%H:%M:%S"
+			#if data_type != "time":
```

Comment:
I've now removed the line entirely since the `time` is now (i.e. after initial implementation where that line was "active") parsed as a `timestamp` (the original value having a `date` prefixed).
I'd left a comment in the code indicating the reason for going this way:
```
# parse Time as a Datetime, since some tests use the ES/SQL-specific
# Time-with-timezone which then needs converting to UTC (as the driver does).
# and this conversion won't work for strptime()'ed Time values, as this uses
# year 1900, not UTC convertible.
```
Thanks for the review.

Reply:
Makes sense, thanks Bogdan.